lib/src/view/design_main_strand.dart

```diff
@@ -352,7 +352,7 @@ assigned, assign the complementary domain names sequence to this strand.
           title: 'set domain name',
           on_click: () => set_domain_name(domain),
         ),
-        if (domain.name != null)
+        if (domain?.name != null)
```

Comment:
I agree this fixes the crash. However, I'm a bit confused under what conditions `domain` will be null. I'm worried that an earlier bug is causing it to be null when it should be not null, and we should be fixing that instead of checking for null here.

Could it be when a 5' or 3' end is clicked? If so, then that seems like a mistake. The component `DesignMainDNAEnd` as a `Domain` as a prop, so I think that domain should be passed as an argument when `context_menu_strand` is called in `on_context_menu` in design_main_strand_dna_end.dart.

Or is there some situation in which `context_menu_strand` is called when there is no `Domain` available associated with the click that caused it to be called?

Reply:
I fixed this in view/design_main_strand_dna_end.dart